engine/src/main/java/org/terasology/particles/ParticleSystemManagerImpl.java

```diff
@@ -0,0 +1,143 @@
+import org.terasology.particles.updating.ParticleUpdater;
+import org.terasology.physics.Physics;
+import org.terasology.registry.In;
+import org.terasology.registry.Share;
+import org.terasology.world.WorldProvider;
+
+import java.util.stream.Stream;
+
+/**
+ * See ParticleSystemManager for more information.
+ */
+@API
+@Share(ParticleSystemManager.class)
+@RegisterSystem(RegisterMode.CLIENT)
+public class ParticleSystemManagerImpl extends BaseComponentSystem implements UpdateSubscriberSystem, ParticleSystemManager {
+
+    private static final Logger logger = LoggerFactory.getLogger(ParticleSystemManagerImpl.class);
+
+    @In
+    private WorldProvider worldProvider;
+
+    @In
+    private EntityManager entityManager;
+
+    @In
+    private Physics physics;
+
+    private ParticleUpdater particleUpdater;
+
+    private BiMap<Class<Component>, GeneratorFunction> registeredGeneratorFunctions = HashBiMap.create();
+    private BiMap<Class<Component>, AffectorFunction> registeredAffectorFunctions = HashBiMap.create();
+
+
+    @ReceiveEvent(components = {ParticleSystemComponent.class})
+    public void onSystemActivated(OnActivatedComponent event, EntityRef entity, ParticleSystemComponent particleSystemComponent) {
+        particleSystemComponent.ownerEntity = entity;
+
+        particleUpdater.register(entity);
+
+        particleUpdater.updateStateData(entity, registeredAffectorFunctions, registeredGeneratorFunctions);
+    }
+
+    @ReceiveEvent(components = {ParticleSystemComponent.class})
+    public void onSystemDeactivated(BeforeDeactivateComponent event, EntityRef entity, ParticleSystemComponent particleSystemComponent) {
+        particleUpdater.dispose(entity);
+    }
+
+    @ReceiveEvent(components = {ParticleSystemComponent.class})
+    public void onSystemChanged(ParticleSystemUpdateEvent event, EntityRef entity, ParticleSystemComponent particleSystemComponent) {
+        particleUpdater.updateStateData(entity, registeredAffectorFunctions, registeredGeneratorFunctions);
+    }
+
+    @ReceiveEvent(components = {ParticleEmitterComponent.class})
+    public void onEmitterActivated(OnActivatedComponent event, EntityRef entity, ParticleEmitterComponent particleEmitterComponent) {
+        particleEmitterComponent.ownerEntity = entity;
+    }
+
+    @ReceiveEvent(components = {ParticleEmitterComponent.class})
+    public void onEmitterChanged(ParticleSystemUpdateEvent event, EntityRef entity) {
+        particleUpdater.updateStateData(entity.getOwner(), registeredAffectorFunctions, registeredGeneratorFunctions);
+    }
+
+
+    public void initialise() {
+        particleUpdater = ParticleUpdater.create(physics);
+    }
+
+    @Override
+    public void shutdown() {
+        registeredAffectorFunctions.clear();
+        registeredGeneratorFunctions.clear();
+
+        particleUpdater = null;
+    }
+
+    public void update(float delta) {
+        particleUpdater.update(delta);
```

Comment:
What's the scenario in which we'd want a particle kill-switch? Disabling all particles at once, irrelevant of what they are used for? I think it makes sense that individual effects can be toggled on/off.

Indeed it is tricky to think about what the desired behavior for particles should be when turning them on/off for a while. For some effects, say a campfire, freezing the particles and simply restarting them from where they left seems reasonable and feasible. For wider effects, say ray, there might be more nuances to that. I.e. if a player moves from an area where it was raining to a desert, reactivating the rain particles should not actually produce any rain. 

Your last question is a bit broad: what do you mean?